--- a/src/fast_livo/CMakeLists.txt
+++ b/src/fast_livo/CMakeLists.txt
@@ -31,8 +31,8 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm|aarch64|ARM|AARCH64)")
   add_definitions(-DARM_ARCH)
 else()
   # x86-64 (Intel/AMD) optimizations
-  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -march=armv8-a -mtune=generic -funroll-loops") #-flto
-  message(STATUS "Using general x86 optimizations: -O3 -march=armv8-a -mtune=generic -funroll-loops") 
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -march=native -mtune=native -funroll-loops") #-flto
+  message(STATUS "Using general x86 optimizations: -O3 -march=native -mtune=native -funroll-loops")
   add_definitions(-DX86_ARCH)
 endif()
